src/ipc/ipc4/handler.c

```diff
@@ -878,6 +878,10 @@ static int ipc4_module_process_dx(struct ipc4_message_request *ipc4)
 		/* do platform specific suspending */
 		platform_context_save(sof_get());
 
+#if !defined(CONFIG_LIBRARY)
+		arch_irq_lock();
+		platform_timer_stop(timer_get());
+#endif
 		ipc_get()->pm_prepare_D3 = 1;
```

Comment:
```c
#if !defined(CONFIG_LIBRARY) && !defined(CONFIG_PM)
```

On MTL  function **_platform_timer_stop_** does nothing from what I've seen.
Interrupts will be disabled later by the PM.

Reply:
@tmleman may I ask why you are suggesting !defined(CONFIG_PM)? Did you mean defined(CONFIG_PM) instead? Would defined(CONFIG_CAVS) be a better option for TGL?

Reply:
**!defined(CONFIG_PM)** and **defined(CONFIG_CAVS)** will have the same result. In my opinion, this code should not go to MTL. It will also probably not be needed once CAVS platforms start using the Zephyr power manager.

**defined(CONFIG_CAVS)** is fine for me too.